src/app/services/user-location.service.ts

```diff
@@ -10,9 +10,11 @@ export class UserLocationService implements OnInit{
   currLocation: any;
   currLat: number;
   currLng: number;
+  storedLat = Number(localStorage.getItem("currLat"));
+  storedLng = Number(localStorage.getItem("currLng"));
 
-  ngOnInit() {
-    this.getLocation();
+  async ngOnInit() {
+    await this.getLocation();
   }
 
   getLocation(){
```

Comment:
an alternative to the async await it to do this: 
if (navigator.geolocation) {
      navigator.geolocation.watchPosition((position) => {
        const pos = {
          lat : position.coords.latitude,
          lng : position.coords.longitude
        }
        this.currLat = pos.lat;
        this.currLng = pos.lng;
        this.currLocation = pos.lat + " " + pos.lng;
      },
      (error) => {
        console.log(error);
      }, 
      { maximumAge: 60000, timeout: 5000, enableHighAccuracy: true });
    } else { 
      alert("Geolocation is not supported by this browser.");
    }